packages/compose/src/hooks/use-drop-zone/index.js

```diff
@@ -107,6 +108,22 @@ export default function useDropZone( {
 				return false;
 			}
 
+			/**
+			 * Checks if the given element is an insertion point.
+			 *
+			 * @param {EventTarget|null} targetToCheck - The element to check.
+			 * @return {boolean} True if the element is a insertion point, false otherwise.
+			 */
+			function isInsertionPoint( targetToCheck ) {
+				const { defaultView } = ownerDocument;
+
+				return !! (
+					defaultView &&
+					targetToCheck instanceof defaultView.HTMLElement &&
+					targetToCheck.dataset.isInsertionPoint
+				);
+			}
```

Comment:
Because this is within the `compose` package we cannot refer to "Zoom out".

Therefore I've aimed to name the concept at work here which is a insertion point between blocks. Hopefully this is suitably generic.

If you apply `data-is-insertion-point` to the element it will be ignored when considering the dragend.

Reply:
Apologies if it's already been explained, but why does this need to happen within the `useDropZone` hook in the `compose` package, rather than within the `onDragLeave` callback in the consumer of the hook? The concept of an insertion point feels bit more conceptually part of the block editor, so looks a bit out of place here to me.

I.e. would it work to perform the insertion point check here instead:

https://github.com/WordPress/gutenberg/blob/f201f8f51d2971969264573327afea0590f82aa0/packages/block-editor/src/components/use-block-drop-zone/index.js#L528-L531

Reply:
Good point. I couldn't see this yesterday as I was very fixed on the solution, but revisiting today it seems obvious that it's better to handle in the consumer rather than low level package.

Interesting, this is the very place I was first able to identify and isolate the original bug so I should have known better 😅